src/CodeReport/Disassembler.cpp

```diff
@@ -24,20 +40,41 @@ void Disassembler::Disassemble(const void* machine_code, size_t size, uint64_t a
   AddLine(absl::StrFormat("Platform: %s",
                           is_64bit ? "X86 64 (Intel syntax)" : "X86 32 (Intel syntax)"));
   err = cs_open(arch, mode, &handle);
-  if (err) {
+  if (err != CS_ERR_OK) {
     AddLine(absl::StrFormat("Failed on cs_open() with error returned: %u", err));
     return;
   }
 
+  cs_option(handle, CS_OPT_DETAIL, CS_OPT_ON);
+
   count = cs_disasm(handle, static_cast<const uint8_t*>(machine_code), size, address, 0, &insn);
 
   if (count) {
     size_t j;
 
     for (j = 0; j < count; j++) {
-      AddLine(
-          absl::StrFormat("0x%llx:\t%-12s %s", insn[j].address, insn[j].mnemonic, insn[j].op_str),
-          insn[j].address);
+      if (IsCallInstruction(insn[j])) {
+        uint64_t callee_address = 0;
+        const orbit_client_protos::FunctionInfo* callee = nullptr;
+        if (absl::numbers_internal::safe_strtou64_base(insn[j].op_str, &callee_address, 16)) {
```

Comment:
Technically we shouldn't call an internal method of Abseil here. (Yeah I know it's frustrating that there is no public API that takes a base argument). You could use `std::from_chars` instead. It's API is almost identical.

Nevertheless, have you considered accessing the operand directly instead of parsing the string representation?

I believe you can check with `cs_op_count` if there is an immediate argument and get with cs_op_index you cang get the corresponding index into the operands array (`insn[j].detail->x86.operands`).

Reply:
Here is some example code: https://github.com/capstone-engine/capstone/blob/master/cstool/cstool_x86.c

Reply:
Thanks! I changed the implementation to check for immediate operands.

Reply:
Very cool. LGTM! Thanks!